src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,340 @@
+    // The rustdoc of the structure.
+    let mut rustdoc = String::new();
+
+    /// For field identifier token, checks if it has already been used then returns it as a
+    /// `String`.
+    macro_rules! check_ident {
+        ($x: ident) => {{
+            if !pre_existing.insert($x) {
+                macro_error!($x, "Identifier already used");
+            }
+        }};
+    }
+
+    /// For bit field indices, checks if they are both within the range of the data type and the
+    /// stop is greater than or equal to the start then return them as `u8`s.
+    macro_rules! index_field {
+        ($start:ident,$stop:ident) => {{
+            // Get start, checking if in range of data type.
+            let start = match $start.to_string().parse::<u8>() {
+                Ok(s) if (0..bits_len.size()).contains(&s) => s,
+                Ok(s) => macro_error!(
+                    $start,
+                    format!(
+                        "Start index ({}) outside valid range ({:?}).",
+                        s,
+                        (0..bits_len.size())
+                    )
+                ),
+                Err(err) => macro_error!(
+                    $start,
+                    format!("Failed to parse token for start index: {}", err)
+                ),
+            };
+            // Get stop, checking if in range of data type.
+            let stop = match $stop.to_string().parse::<u8>() {
+                Ok(s) if (start..=bits_len.size()).contains(&s) => s,
+                Ok(s) => macro_error!(
+                    $stop,
+                    format!(
+                        "Stop index ({}) outside valid range ({:?}).",
+                        s,
+                        (start..=bits_len.size())
+                    )
+                ),
+                Err(err) => macro_error!(
+                    $stop,
+                    format!("Failed to parse token for stop index: {}", err)
+                ),
+            };
+            (start, stop)
+        }};
+    }
+    /// For bit flag index, checks if it is within the range of the data type,
+    /// then returns it as a `u8`.
+    macro_rules! index_flag {
+        ($index:ident) => {{
+            // Get index, checking if in range of data type.
+            match $index.to_string().parse::<u8>() {
+                Ok(s) if (0..bits_len.size()).contains(&s) => s,
+                Ok(s) => macro_error!(
+                    $index,
+                    format!(
+                        "Index ({}) outside of valid range ({:?}).",
+                        s,
+                        (0..bits_len.size())
+                    )
+                ),
+                Err(err) => {
+                    macro_error!($index, format!("Failed to parse token for index: {}", err))
+                }
+            }
+        }};
+    }
+    // Iterate through the token stream, matching slices to:
+    // 1. Bit fields.
+    // 2. Bit flags.
+    // 3. Rustdoc comments.
```

Comment:
Maybe I am missing some details, but I do not understand the rustdoc part.
For the sake of simplicity of code and maintainability purposes, couldn t we just not suport rust doc comments?
If I am not mistaken, the we will be the ones dictating the format of the cpu templates. 
Why support comments in them?

Reply:
We could drop support for Rustdoc comments in the macro, but it doesn't simplify it much and I felt it could be a  worthwhile inclusion, for present and future documentation.